zerver/views/users.py

```diff
@@ -297,12 +302,14 @@ def add_bot_backend(request, user_profile, full_name_raw=REQ("full_name"), short
         user_file = list(request.FILES.values())[0]
         upload_avatar_image(user_file, user_profile, bot_profile)
 
-    if bot_type == UserProfile.OUTGOING_WEBHOOK_BOT:
-        add_outgoing_webhook_service(name=service_name,
-                                     user_profile=bot_profile,
-                                     base_url=payload_url,
-                                     interface=1,
-                                     token=random_api_key())
+    if bot_type == UserProfile.OUTGOING_WEBHOOK_BOT or bot_type == UserProfile.EMBEDDED_BOT:
+        if bot_type == UserProfile.EMBEDDED_BOT:
+            payload_url = ""
+        add_service(name=service_name,
+                    user_profile=bot_profile,
+                    base_url=payload_url,
+                    interface=1,
+                    token=random_api_key())
```

Comment:
I have already added the check whether the bot is in our registry or not, in the PR you just merged [here](https://github.com/zulip/zulip/pull/5882/files#diff-10573c5f0fd3eebaa30a105a7177982aR30)

I'll have to add the error handling part.

Well, the fact that I did not give much attention to checking this is because the users can only select the services through the dropdown services listed. That is why I added just one check.